### PR TITLE
fixes the way the informers are started in sample controller pkg

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: there is no need to run `informerFactory.Start` method in a separate `goroutine`. The method is non-blocking and runs all registered informers in a dedicated `goroutine`.

**Release note**:

```
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/sample-controller/main.go [hunk 1]
```diff
@@ -65,8 +65,8 @@ func main() {
 		kubeInformerFactory.Apps().V1().Deployments(),
 		exampleInformerFactory.Samplecontroller().V1alpha1().Foos())
 
-	go kubeInformerFactory.Start(stopCh)
-	go exampleInformerFactory.Start(stopCh)
+	kubeInformerFactory.Start(stopCh)
+	exampleInformerFactory.Start(stopCh)
```
Comment (nikhita) [inline staging/src/k8s.io/sample-controller/main.go]: :+1:

File: staging/src/k8s.io/sample-controller/main.go [hunk 2]
```diff
@@ -65,8 +65,8 @@ func main() {
...
 
 	if err = controller.Run(2, stopCh); err != nil {
```
Comment (nikhita) [inline staging/src/k8s.io/sample-controller/main.go]: Shouldn't this be run as a go routine though?

Comment (p0lyn0mial) [reply staging/src/k8s.io/sample-controller/main.go]: I think it is okay, the main func will stop at this method as it essentially runs the controller and blocks until the stopCh is closed.

File: staging/src/k8s.io/sample-controller/main.go (hunk 1 above)
Comment (p0lyn0mial) [reply staging/src/k8s.io/sample-controller/main.go]: sgtm

**Comment by p0lyn0mial** [comment]
added the comment. PTAL

**Pull request merged**